kyo-prelude/shared/src/test/scala/kyo2/StreamTest.scala

```diff
@@ -0,0 +1,650 @@
+            )
+        }
+    }
+
+    "initChunk" - {
+        "empty" in {
+            assert(
+                Stream.init(Chunk.empty[Int]).runSeq.eval == Seq.empty
+            )
+        }
+
+        "non-empty" in {
+            assert(
+                Stream.init(Chunk(1, 2, 3)).runSeq.eval == Seq(1, 2, 3)
+            )
+        }
+    }
+
+    "take" - {
+        "zero" in {
+            assert(
+                Stream.init(Seq(1, 2, 3)).take(0).runSeq.eval == Seq.empty
+            )
+        }
+
+        "negative" in {
+            assert(
+                Stream.init(Seq(1, 2, 3)).take(-1).runSeq.eval == Seq.empty
+            )
+        }
+
+        "two" in {
+            assert(
+                Stream.init(Seq(1, 2, 3)).take(2).runSeq.eval == Seq(1, 2)
+            )
+        }
+
+        "more than available" in {
+            assert(
+                Stream.init(Seq(1, 2, 3)).take(5).runSeq.eval == Seq(1, 2, 3)
+            )
+        }
+
+        "stack safety" in {
+            assert(
+                Stream.init(Seq.fill(100000)(1)).take(5).runSeq.eval ==
+                    Seq.fill(5)(1)
+            )
+        }
+    }
+
+    "drop" - {
+        "zero" in {
+            assert(
+                Stream.init(Seq(1, 2, 3)).drop(0).runSeq.eval == Seq(1, 2, 3)
+            )
+        }
+
+        "negative" in {
+            assert(
+                Stream.init(Seq(1, 2, 3)).drop(-1).runSeq.eval == Seq(1, 2, 3)
+            )
+        }
+
+        "two" in {
+            assert(
+                Stream.init(Seq(1, 2, 3)).drop(2).runSeq.eval == Seq(3)
+            )
+        }
+
+        "more than available" in {
+            assert(
+                Stream.init(Seq(1, 2, 3)).drop(5).runSeq.eval == Seq.empty
+            )
+        }
+
+        "stack safety" in {
+            assert(
+                Stream.init(Seq.fill(n)(1)).drop(5).runSeq.eval.size == n - 5
+            )
+        }
+    }
+
+    "takeWhile" - {
+        "take none" in {
+            assert(
+                Stream.init(Seq(1, 2, 3)).takeWhile(_ < 0).runSeq.eval == Seq.empty
+            )
+        }
+
+        "take some" in {
+            assert(
+                Stream.init(Seq(1, 2, 3, 4, 5)).takeWhile(_ < 4).runSeq.eval ==
+                    Seq(1, 2, 3)
+            )
+        }
+
+        "take all" in {
+            assert(
+                Stream.init(Seq(1, 2, 3)).takeWhile(_ < 10).runSeq.eval ==
+                    Seq(1, 2, 3)
+            )
+        }
+
+        "empty stream" in {
+            assert(
+                Stream.init(Seq.empty[Int]).takeWhile(_ => true).runSeq.eval ==
+                    Seq.empty
+            )
+        }
+
+        "with effects" in {
+            val stream = Stream.init(Seq(1, 2, 3, 4, 5))
+            val taken = stream.takeWhile { v =>
+                Var.update[Int](_ + 1).map(_ < 4)
+            }.runSeq
+            assert(Var.run(0)(taken).eval == Seq(1, 2, 3))
```

Comment:
Can you also validate the result of the Var? This way we can check that take actually stops the stream when receiving the ack.